### DISCUSSION TITLE
Few newbie issues/errors with BlenderBIM

### DISCUSSION THREAD

**Discussion by Hexodus1** [IfcOpenShell/IfcOpenShell#2779] [Q&A] [answered]
Hi there! I'm very new to the BlenderBIM and OpenBIM in general, but really stoked about the concept and its possibilities. I've been doing a little experiment modeling an existing building from scratch, with just a floor plan to start with. These are the few issues I've run into. I've been using the newest stable version 0.0.230107.

1. Upon saving/opening the IFC Project Library (I've been using the IFC4 Demo Library) seems to disappear. I've been saving both the IFC project file and the Blender file (.blend) so that the reference image is included in the save. Should I save/open project using only the IFC "Save Project" function?

2. Sometimes (also might have something to do with the saving but this happens also randomly) the objects lose their IFC class. I have  now random walls in the project that don't show any IFC object metadata and also can't be adjusted (height/lenght/etc.) anymore.

![BlenderBIM_wall_adjust_error](https://user-images.githubusercontent.com/125542628/219321775-99df73c1-371d-43a3-a7a6-e22aba591daf.jpg)

3. An error when trying to save the IFC-project file. This didn't occur until I had saved multiple different versions of the project file. This may have something to do with the blender file save also.

![BlenderBIM_IFC_project_save_error](https://user-images.githubusercontent.com/125542628/219321767-5e37fee1-f602-47dc-8da9-84b55d628e9d.jpg)

I know these are probably all just user errors trying to use the software in a way it was not intended. I've tried to recognize any patterns what might cause these errors but apart from the saving (blender file) I can't seem to find any. If there are any thorough tutorials that could help avoiding these errors, please let me know.

Thanks in advance!

**Comment by Moult** [comment, marked as answer]
For these types of errors, it's almost always _not_ user error, and instead it's a bug which if you do manage to isolate (tip: save early, save often, be very aware of every button you press or action you take, and try to recreate the scenario and post a bug). A few things to keep in mind:

 1. In general, always use the latest unstable release, which is actually uh... more stable than the stable release :) Except when it's not, but in more cases it's more stable. Like significantly so.
 2. When an object loses their class, it is a critical error and occurs when the object in Blender references an IFC element that has been removed. This is very easy to replicate, by first saving the IFC, then changing something in Blender, then somehow saving the Blender file but not saving the IFC, or swapping out the IFC with another IFC, etc. Remember: the IFC holds all the data, the Blender session is simply a "working session" or "cache" of what Blender thinks is in the IFC. To be very safe, never use .blend files at all, and always use the "Save Project" button to save to IFC. That's what I do. It's not ideal, but clearly we have enough .blend "not synced" bugs that have not been solved yet. These are the hardest bugs to find and fix but over time it should increasingly get more stable.


Hopefully that helps explain some of your symptoms. Please really do help test and report bugs. They are high priority critical issues to fix.

**Comment by Hexodus1** [comment]
Thank you for the quick response! This was very helpful and also encouraging to try to isolate the bugs and report them. I will definitely keep exploring Blender BIM and its many possibilities. Cheers!
